src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_emitter.cpp

```diff
@@ -70,54 +88,56 @@ std::set<std::vector<element::Type>> jit_brgemm_emitter::get_supported_precision
 }
 
 void jit_brgemm_emitter::validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const {
-    OV_CPU_JIT_EMITTER_ASSERT((m_with_scratch && in.size() == 3) || (!m_with_scratch && in.size() == 2),
+    OV_CPU_JIT_EMITTER_ASSERT(m_memory_offsets.size() == in.size() + 1 && (out.size() == 1),
                               "expects 3 inputs if there are compensations/wsp");
 }
 
 void jit_brgemm_emitter::emit_impl(const std::vector<size_t>& in, const std::vector<size_t>& out) const {
     validate_arguments(in, out);
     if (host_isa_ == cpu::x64::avx512_core) {
-        Xbyak::Reg64 input_0(static_cast<int>(in[0]));
-        Xbyak::Reg64 input_1(static_cast<int>(in[1]));
-        Xbyak::Reg64 input_2(static_cast<int>(m_with_scratch ? in[2] : 0));  // scratch. Default reg index is 0 if there isn't scratch
-        Xbyak::Reg64 output_0(static_cast<int>(out[0]));
-        emit_brgemm_kernel_call(input_0,
-                                input_1,
-                                input_2,
-                                output_0,
-                                m_load_offset_a,
-                                m_load_offset_b,
-                                m_load_offset_scratch,
-                                m_store_offset_c);
+        std::vector<size_t> mem_ptrs_idxs{in[0], in[1], out[0]};
+        if (in.size() > 2)
+            mem_ptrs_idxs.emplace_back(in[2]);
+        emit_brgemm_kernel_call(mem_ptrs_idxs, m_memory_offsets);
     } else {
         OV_CPU_JIT_EMITTER_THROW("requires at least avx512_core instruction set");
     }
 }
-
-void jit_brgemm_emitter::emit_brgemm_kernel_call(Reg64 addr_A, Reg64 addr_B, Reg64 scratch, Reg64 addr_C,
-                                                 const size_t in0_kernel_offset, const size_t in1_kernel_offset,
-                                                 const size_t in2_kernel_offset, const size_t out0_kernel_offset) const {
+void jit_brgemm_emitter::emit_brgemm_kernel_call(const std::vector<size_t>& mem_ptrs_idxs, const std::vector<size_t>& mem_offsets) const {
     internal_call_preamble();
     h->mov(h->rbp, reinterpret_cast<uint64_t>(BrgemmKernelExecutor::execute));
     auto reserved_stack_size = sizeof(BrgemmKernelExecutor::call_args);
     // Reserve memory on the stack
     h->sub(h->rsp, reserved_stack_size);
 
-    auto write_addr_on_stack = [&](size_t arg_offset, Reg64 addr, size_t addr_offset) {
+    Xbyak::Reg64 aux_reg = [this, &mem_ptrs_idxs]() {
+        std::set<size_t> used(mem_ptrs_idxs.begin(), mem_ptrs_idxs.end());
+        std::vector<Xbyak::Reg64> spilled_gprs {h->r8, h->r9, h->r10, h->r11, h->r12, h->r13, h->r14, h->r15,
```

Comment:
Minor: maybe make it static variable?

Reply:
We can't really make the `spilled_regs` static, since they depend on `dnnl::impl::cpu::x64::jit_generator* h` which is a class field that needs to be initialized in the `jit_emitter` constructor.
In theory, we can create a `std::vector<Reg64> jit_emitter::get_spilled_gprs() const`, but this will affect the interface of the widely used class, and this feature is really needed in just two places. 
I agree that this code repetition is annoying, but we probably need a better justification to alter the base class (e.g. more usages)